### PR TITLE
feat: 438 vep annotation order

### PR DESCRIPTION
**Pull request by ericblanc20**
Impose user-defined annotation order for VEP.
Previously, the wrapper was using the default order, based on ENSEMBL canonical transcript.
Because of that choice, many variant were not uploaded to cBioPortal.

The default choice now prioritizes protein-coding genes with MANE transcripts.
Many variants are now recovered in cBioPortal views.

### CODE REVIEW

File: snappy_pipeline/workflows/somatic_variant_annotation/__init__.py [hunk 1]
```diff
@@ -26,7 +26,31 @@
 Step Output
 ===========
 
-TODO
+Annotations can be done on all genes & transcripts overlapping with the variant locus, or
+they can select one representative gene and transcript for annotation.
+In the latter case, the output vcf file will only contain one annotation per variant, while
+in the former case, there might be over 100 annotations for each variant.
+
+The ordering of features drinving the representative annotation choice is under user control.
```
Comment (mbenary) [inline snappy_pipeline/workflows/somatic_variant_annotation/__init__.py:34]: Typo: 

> driving

File: snappy_wrappers/wrappers/vep/post_filter/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        for criterion in criteria:
+            result = get_value(criterion, fields[criterion])
+            if result is None or result == codes[criterion]:
+                continue
+            selected = result < codes[criterion]
+            break
+
+        if selected:
+            for criterion in criteria:
+                codes[criterion] = get_value(criterion, fields[criterion])
```
Comment (mbenary) [inline snappy_wrappers/wrappers/vep/post_filter/wrapper.py:171]: Probably a very simple thing: Why is codes changed here? It's not used afterwards.

**Review by mbenary — CHANGES_REQUESTED** [review]
Just a few tiny things, sorry for being picky.

File: snappy_pipeline/workflows/somatic_variant_annotation/__init__.py [hunk 3]
```diff
@@ -115,7 +139,7 @@
       assembly: GRCh38
       cache_version: 102        # WARNING- this must match the wrapper's vep version!
       tx_flag: "gencode_basic"  # The flag selecting the transcripts.  One of "gencode_basic", "refseq", and "merged".
-      pick: yes                 # Other option: no (report one or all consequences)
+      pick_order: ["biotype", "mane", "appris", "tsl", "ccds", "canonical", "rank", "length"]
```
[the remark this answers is not in the document]
Comment (ericblanc20) [reply snappy_pipeline/workflows/somatic_variant_annotation/__init__.py:142]: This is the default order. In `check_config`, the code tests validity of all the options provided by the user in the yaml file

**Review by mbenary — APPROVED** [review]
LGTM

**Pull request merged**